### PR TITLE
instance of

### PR DESCRIPTION
**Pull request by sreechand2002**
checking instance of

### CODE REVIEW

File: src/main/java/pinger/Pinger.java [hunk 1]
```diff
@@ -18,19 +18,24 @@ public static void main(String[] args)  {
                 String urlI = "http://" + reader.readLine();
                 URL url = new URL(urlI);
                 HttpURLConnection conn = (HttpURLConnection) url.openConnection();
```
Comment (BoogieMonster1O1) [inline src/main/java/pinger/Pinger.java]: You also need to remove the cast here. Change this to a `URLConnection` and create another variable inside the `if` statement which does the cast.

**Review by BoogieMonster1O1 — CHANGES_REQUESTED** [review]

**Review by sreechand2002 — COMMENTED** [review]
changed it

File: src/main/java/pinger/Pinger.java [hunk 2]
```diff
@@ -17,20 +17,25 @@ public static void main(String[] args)  {
...
-                    if (content.isEmpty()) {
-                        content = "No content returned";
+                URLConnection conn = url.openConnection();
+                if(conn instanceof HttpURLConnection) {
+                    try (InputStream stream = conn.getInputStream()) {
+                        String content = readAllLines(stream);
+                        if (content.isEmpty()) {
+                            content = "No content returned";
+                        }
+                        addField("Content Type", conn.getContentType());
```
Comment (BoogieMonster1O1) [inline src/main/java/pinger/Pinger.java]: This doesn't compile.
You should make another variable, say `conn1`, and assign it to `(HttpURLConnection) conn`.

**Review by BoogieMonster1O1 — CHANGES_REQUESTED** [review]

**Comment by sreechand2002** [comment]
now complies successfully

**Review by BoogieMonster1O1 — APPROVED** [review]
LGTM

**Comment by Rishabkashyap14** [comment]
!bounty 100

**Pull request merged**